src/libstd/collections/hash/map.rs

```diff
@@ -95,53 +95,61 @@ fn test_resize_policy() {
     }
 }
 
-// The main performance trick in this hashmap is called Robin Hood Hashing.
-// It gains its excellent performance from one essential operation:
+// The main performance trick in this hashmap is called Robin Hood linear
+// probing hashing. It gains its excellent performance from one essential
+// operation:
 //
 //    If an insertion collides with an existing element, and that element's
-//    "probe distance" (how far away the element is from its ideal location)
-//    is higher than how far we've already probed, swap the elements.
+//    "displacement" (how far away the element is from its ideal location)
+//    is lower than how far we've already probed, swap the elements.
 //
-// This massively lowers variance in probe distance, and allows us to get very
+// This massively lowers variance in displacement, and allows us to get very
 // high load factors with good performance. The 90% load factor I use is rather
 // conservative.
 //
 // > Why a load factor of approximately 90%?
 //
 // In general, all the distances to initial buckets will converge on the mean.
-// At a load factor of α, the odds of finding the target bucket after k
-// probes is approximately 1-α^k. If we set this equal to 50% (since we converge
-// on the mean) and set k=8 (64-byte cache line / 8-byte hash), α=0.92. I round
-// this down to make the math easier on the CPU and avoid its FPU.
-// Since on average we start the probing in the middle of a cache line, this
-// strategy pulls in two cache lines of hashes on every lookup. I think that's
-// pretty good, but if you want to trade off some space, it could go down to one
-// cache line on average with an α of 0.84.
+// At a load factor of α, the odds of finding the target bucket after exactly n
+// unsuccesful probes[3] are
 //
-// > Wait, what? Where did you get 1-α^k from?
+// Pr{displacement = n} =
+// (1 - α) / α * ∑_{k≥1} e^(-kα) * (kα)^(k+n) / (k + n)! * (1 - kα / (k + n + 1))
 //
-// On the first probe, your odds of a collision with an existing element is α.
-// The odds of doing this twice in a row is approximately α^2. For three times,
-// α^3, etc. Therefore, the odds of colliding k times is α^k. The odds of NOT
-// colliding after k tries is 1-α^k.
+// These odds can be approximated with this code:
 //
-// The paper from 1986 cited below mentions an implementation which keeps track
-// of the distance-to-initial-bucket histogram. This approach is not suitable
-// for modern architectures because it requires maintaining an internal data
-// structure. This allows very good first guesses, but we are most concerned
-// with guessing entire cache lines, not individual indexes. Furthermore, array
-// accesses are no longer linear and in one direction, as we have now. There
-// is also memory and cache pressure that this would entail that would be very
-// difficult to properly see in a microbenchmark.
```

Comment:
It's not relevant enough. I figured out that the paper discusses two different strategies.